src/shogun/regression/KernelRidgeRegression.cpp

```diff
@@ -138,18 +81,21 @@ bool CKernelRidgeRegression::train_machine(CFeatures *data)
 	}
 	ASSERT(kernel && kernel->has_features())
 
-	switch (m_train_func)
+	if (m_labels->get_num_labels() != kernel->get_num_vec_rhs())
```

Comment:
@karlnapf if you look at line [73](https://github.com/shogun-toolbox/shogun/pull/3098/files#diff-429ee3465120e3c46c4432c53977a7cfR73), the first check can't go in setters/constructors for this use-case:
```
krr = KernelRidgeRegression()
krr.set_kernel(kernel)
krr.set_labels(labels)
krr.train()
```
The second check is explicit for krr. Last check can only be performed when `m_labels` is not NULL.